src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs

```diff
@@ -63,6 +63,104 @@ public static void BuildChain()
+                Assert.Equal(microsoftDotComIssuer, chain2.ChainElements[1].Certificate);
+                Assert.Equal(microsoftDotComRoot, chain2.ChainElements[2].Certificate);
+
+                chain2.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
+                chain2.ChainPolicy.VerificationTime = new DateTime(2015, 10, 15, 12, 01, 01, DateTimeKind.Local);
+                chain2.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+                valid = chain2.Build(microsoftDotCom);
+                Assert.True(valid, "Chain built validly");
+            }
+
+            // No exception thrown for accessing ChainContext on disposed chain
+            Assert.Equal(IntPtr.Zero, chain.ChainContext);
+        }
+
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [Fact]
+        public static void VerifyChainFromHandle_Unix()
+        {
+            using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
+            using (var chainHolder = new ChainHolder())
+            {
+                X509Chain chain = chainHolder.Chain;
+                chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
+                chain.ChainPolicy.VerificationTime = new DateTime(2015, 10, 15, 12, 01, 01, DateTimeKind.Local);
+                chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+
+                bool valid = chain.Build(microsoftDotCom);
+
+                Assert.Equal(IntPtr.Zero, chain.ChainContext);
+            }
+
+            Assert.Throws<PlatformNotSupportedException>(() => new X509Chain(IntPtr.Zero));
+        }
+
+        [Fact]
+        public static void TestResetMethod()
+        {
+            using (var sampleCert = new X509Certificate2(TestData.DssCer))
+            using (var chainHolder = new ChainHolder())
+            {
+                X509Chain chain = chainHolder.Chain;
+
+                chain.ChainPolicy.ExtraStore.Add(sampleCert);
+                bool valid = chain.Build(sampleCert);
+                Assert.False(valid);
+
+                chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
+                chain.ChainPolicy.VerificationTime = new DateTime(2015, 10, 15, 12, 01, 01, DateTimeKind.Local);
+                chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+
+                valid = chain.Build(sampleCert);
+                Assert.True(valid, "Chain built validly");
+
+                Assert.Equal(1, chain.ChainElements.Count);
+
+                chain.Reset();
+                Assert.Equal(0, chain.ChainElements.Count);
+
+                chain.ChainPolicy.ExtraStore.Add(sampleCert);
+                valid = chain.Build(sampleCert);
+                Assert.Equal(1, chain.ChainElements.Count);
+
+                // This succeeds because ChainPolicy is not reset in Reset() for desktop compat
```

Comment:
How about actually asserting that, instead of just comment asserting it?  It's indirectly tested via building the chain, but it'd be pretty annoying to track down the root cause from that failure.